packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -661,21 +661,34 @@ function styleTagPropsFromRawProps(
 }
 
 function getStyleKey(href: string) {
-  const limitedEscapedHref =
-    escapeSelectorAttributeValueInsideDoubleQuotes(href);
-  return `href~="${limitedEscapedHref}"`;
+  return href;
 }
 
 function getStyleTagSelectorFromKey(key: string) {
-  return `style[data-${key}]`;
+  // Key is actually the href
+  // @TODO refactor. It was convient before when the key being the selector was sufficient
+  // but with the complexities introduced with <style> support this structure makes less sense.
+  const limitedEscapedHref =
+    escapeStyleHrefAttributeValueInsideDoubleQuotes(key);
+  return `style[data-href~="${limitedEscapedHref}"]`;
```

Comment:
This whitelist separated selector is the crux of the issues of the latest commit. We can't tolerate spaces in the url because this selector will fail otherwise so we either need to require your to encode your space containing URLs (which you should) except that these are probably not always going to be real URLs (css-in-js) and someone out there will accidentally include a space and break hydration on the client.

To resolve that we (for `<style>` only) map `' '` to `%20`. This works for now but it introduces a problem with css inlining

In the future if we inline, we might take multiple hrefs for stylesheets and inline them into a single `<style>` tag.

Later when we get our selector for stylesheets we (currently) produce one that does not escape spaces. It is unlikely folks will use spaces-in-urls since they are invalid characters but if one did the style would have escaped the space before emitting the href and the selector wouldn't know to. We could special case it and create one selector for finding `<link ...` and another for finding `<style ...`

Either way that can maybe be a problem for when we implement inlining

Reply:
It’s a non-issue. Just make css in js libraries encode space. It’ll be a hash or something anyway so they won’t have a hash.